### ISSUE TITLE
Question about using SigLIP loss in FG-CLIP

### ISSUE THREAD

**Issue by Link7808** [360CVGroup/FG-CLIP#48]
Hi, thanks for the great work! I’m curious whether you have explored using a SigLIP-style loss (sigmoid-based contrastive loss) instead of the standard CLIP InfoNCE in FG-CLIP. If so, any observations on performance or stability? Thanks!

**Comment by binwang777** [comment]
> Hi, thanks for the great work! I’m curious whether you have explored using a SigLIP-style loss (sigmoid-based contrastive loss) instead of the standard CLIP InfoNCE in FG-CLIP. If so, any observations on performance or stability? Thanks!

FG-CLIP2’s global loss adopts a SigLIP-style loss (sigmoid-based contrastive loss). Please refer to the loss implementation in `https://github.com/360CVGroup/FG-CLIP/blob/main/fgclip2/model/strcs/fgclip2.py`:

```
def all_reduce_siglip_loss(self, image_features, text_features, logit_scale, logit_bias, cur_rank, no_longtext_indices=None, output_dict=False):
        
        loss = self._loss(image_features, text_features, logit_scale, logit_bias)


        for i in range(self.world_size):

            text_from_other = torch.distributed.nn.all_reduce(
                text_features * (cur_rank == i),
                torch.distributed.ReduceOp.SUM,
            )

            loss += float(i != cur_rank) * self._loss(
                image_features,
                text_from_other,
                logit_scale,
                logit_bias,
                negative_only=True,
            )

        return loss
```

The code uses an `all_reduce` operator to gather samples from the other ranks for loss computation. Before using this loss, you need to set the total number of GPUs you are using, i.e., `world_size`. Although `torch.distributed.ReduceOp.SUM` might make it seem like a simple summation, you can verify this logic on a local multi-GPU machine—it indeed incorporates samples from the other ranks.

**Comment by Link7808** [comment]
Thanks for the reply!
